tools-local/unix/cmake/get-version.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# Gets the version of CMake at the specified path.
+
+usage()
+{
+    echo "Usage: $0 cmake ToolName --ToolPath <path>"
```

Comment:
> We'd only need a single place for named args, then: build-native.sh, where there's already an arg parse loop.

Glad you brought this up.  Yes, `probe-tool` gets required, positional arguments only.  Since `probe-tool` invokes the extension, it can determine which arguments to pass to the extension. For example, `acquire-tool` extension needs toolname and override-tools-folder-path. `acquire-tool` does not care about strict-tool-version-match. 
Required arguments for probe-tool are:

1. Name of the tool
2. Strict tool version match. Empty string to indicate strict tool version matching is not required. "strict" to ensure version matches declared version
3. Override scripts folder path. Empty string to indicate build did not specify an override scripts path.

Similarly, other scripts can be designed to get required args, meaning no named args.